src/FakeObject.php

```diff
@@ -1,69 +1,80 @@
 <?php
+
+namespace Chillu\FakeDatabase;
+
+use \ArrayObject;
+
 /**
  * Simple key/value map with defaults support.
  */
-class FakeObject extends ArrayObject {
+class FakeObject extends ArrayObject
+{
 
-	function __construct($data = null, $flags = ArrayObject::ARRAY_AS_PROPS, $iteratorClass = 'ArrayIterator') {
-		if(!$data) $data = array();
-		parent::__construct($data, $flags, $iteratorClass);
+    public function __construct($data = null, $flags = ArrayObject::ARRAY_AS_PROPS, $iteratorClass = 'ArrayIterator')
+    {
+        if (!$data) {
+            $data = array();
+        }
+        
+        parent::__construct($data, $flags, $iteratorClass);
 
-		foreach($this->getDefaults() as $k => $v) {
-			if(!isset($this[$k])) $this[$k] = $v;
-		}
-	}
-	
-	public function getDefaults() {
-		return array();
-	}
-	
-	/**
-	 * Serialize object and contained objects an array,
-	 * in a format with "_type" hints, useful for later restoring
-	 * through {@link create_from_array()}.
-	 * 
-	 * @see http://stackoverflow.com/questions/6836592/serializing-php-object-to-json
-	 * @return Array
-	 */
-	public function toArray() {
-		$array = (array)$this;
-		$array['_type'] = get_class($this);
-		array_walk_recursive($array, function(&$property, $key){
-			if($property instanceof FakeObject){
-				$property = $property->toArray();
-			}
-		});
-		return $array;
-	}
+        foreach ($this->getDefaults() as $k => $v) {
+            if (!isset($this[$k])) {
+                $this[$k] = $v;
+            }
+        }
+    }
+    
+    public function getDefaults()
+    {
+        return array();
+    }
+    
+    /**
+     * Serialize object and contained objects an array,
+     * in a format with "_type" hints, useful for later restoring
+     * through {@link createFromArray()}.
+     *
+     * @see    http://stackoverflow.com/questions/6836592/serializing-php-object-to-json
+     * @return array
+     */
+    public function toArray()
+    {
+        $array = (array)$this;
+        $array['_type'] = get_class($this);
+        
+        array_walk_recursive(
+            $array,
+            function (&$property, $key) {
+                if ($property instanceof FakeObject) {
+                    $property = $property->toArray();
+                }
+            }
+        );
+        
+        return $array;
+    }
 
-	/**
-	 * Create nested object representation from array,
-	 * based on "_type" hints.
-	 *
-	 * @param Array
-	 * @return FakeObject
-	 */
-	public static function create_from_array($array) {
-		// array_walk_recursive doesn't recurse into arrays...
-		foreach($array as &$v) {
-			// Convert "has one" relations
-			if(is_array($v)) {
-				$v = FakeObject::create_from_array($v);
-			} 
-			// Convert "has many" relations
```

Comment:
lol so the `elseif` is the same condition as the `if`. There's no test coverage for `has_many` or `many_many`. Is this a feature regression?

Reply:
Yeah I couldn't work out (nor remember) what this was supposed to be for anyway. As far as I could figure from the tests everything was covered by either `not array` or `is array` and I couldn't work out what the comments 'has one' and 'has many' implied (except for the distinction between 'it's a single value' and 'it's an array of values' to create)...

Can you remember what this was supposed to be used for?

Reply:
Nope, if the tests pass, and your use case for the project you're migrating is still satisfied, I'm happy to leave it as-is

Reply:
Yeah I couldn't work out (nor remember) what this was supposed to be for anyway. As far as I could figure from the tests everything was covered by either `not array` or `is array` and I couldn't work out what the comments 'has one' and 'has many' implied (except for the distinction between 'it's a single value' and 'it's an array of values' to create)...

Can you remember what this was supposed to be used for?

Reply:
Cool - yeah all tests pass and I can't figure out what it's used for. If I ever do, I can re-instate something :)